java/client/src/org/openqa/selenium/remote/RemoteWebDriverBuilder.java

```diff
@@ -174,261 +191,260 @@ public RemoteWebDriverBuilder url(String url) {
    * {@link #withDriverService(DriverService)}.
    */
   public RemoteWebDriverBuilder url(URL url) {
-    this.remoteHost = Require.nonNull("Remote server URL", url);
-    validateDriverServiceAndUrlConstraint();
-    return this;
+    Require.nonNull("Remote server URL", url);
+    try {
+      return uri(url.toURI());
+    } catch (URISyntaxException e) {
+      throw new IllegalArgumentException("Unable to create new URI from " + url, e);
+    }
   }
 
   /**
-   * Use the given {@link DriverService} to set up the webdriver instance. It is an error to set
-   * both this and also call {@link #url(URL)}.
+   * @see #uri(URI)
    */
-  public RemoteWebDriverBuilder withDriverService(DriverService service) {
-    this.service = Require.nonNull("Driver service", service);
-    validateDriverServiceAndUrlConstraint();
-    return this;
+  public RemoteWebDriverBuilder uri(String uri) {
```

Comment:
Most people are already trained to use `URL`s and existing code does that for the `RemoteWebDriver` constructor. However, to use different schemes (notably `ws`, `wss` and `unix`) one needs to use `URI` instead, since `URL` does a check to see if there's a handler for each protocol that `URI` omits.

I'll pick a name that encapsulates both concepts.